core/src/main/java/ai/timefold/solver/core/config/heuristic/selector/move/MoveSelectorConfig.java

```diff
@@ -234,28 +237,26 @@ public Config_ withFixedProbabilityWeight(Double fixedProbabilityWeight) {
      * Gather a list of all descendant {@link MoveSelectorConfig}s
      * except for {@link UnionMoveSelectorConfig} and {@link CartesianProductMoveSelectorConfig}.
      *
-     * @param leafMoveSelectorConfigList not null
      */
-    public void extractLeafMoveSelectorConfigsIntoList(List<MoveSelectorConfig> leafMoveSelectorConfigList) {
+    // TODO: list elements @NonNull?
+    public void extractLeafMoveSelectorConfigsIntoList(@NonNull List<@NonNull MoveSelectorConfig> leafMoveSelectorConfigList) {
```

Comment:
@triceo There are multiple `List<T>` member in the `*Config`-classes. My current assumption was that `null` does not make a whole lot of sense here, so I annotated the type param as `@NonNull`. Is this assumption valid for you?

Reply:
Yes, it makes sense. I would not expect null inside of this collection.